### PR TITLE
fix: M2-3518 Fix activity flow navigation issue

### PR DESCRIPTION
**Pull request by IhorNesterenko99**

### CODE REVIEW

File: src/modules/Builder/hooks/useActivityFlowsRedirection.ts [hunk 1]
```diff
@@ -14,16 +14,17 @@ export const useActivityFlowsRedirection = () => {
...
 
-  const activities = watch('activities');
-  const activityFlows = watch('activityFlows');
-  const activityFlow = activityFlows?.find(
-    (flow: ActivityFlowFormValues) => activityFlowId === getEntityKey(flow),
-  );
-  const shouldRedirect = (isNewApplet || activities?.length > 0) && activityFlowId && !activityFlow;
-
   useEffect(() => {
+    const activities = watch('activities');
```
Comment (kdanilenka32) [inline src/modules/Builder/hooks/useActivityFlowsRedirection.ts:18]: Is it ok watch inside useEffect?

Comment (IhorNesterenko99) [reply src/modules/Builder/hooks/useActivityFlowsRedirection.ts:18]: there are no errors, it seems it's just a function

Comment (kdanilenka32) [reply src/modules/Builder/hooks/useActivityFlowsRedirection.ts:18]: if use getValues instead of create subscriptions

Comment (alex-i-e) [reply src/modules/Builder/hooks/useActivityFlowsRedirection.ts:18]: please don't unsubscribe then 
  // Callback version of watch.  It's your responsibility to unsubscribe when done.
  React.useEffect(() => {
    const subscription = watch((value, { name, type }) => console.log(value, name, type));
    return () => subscription.unsubscribe();
  }, [watch]);

**Review by alex-i-e — CHANGES_REQUESTED** [review]
please don't forget to unsubscribe then 
https://www.react-hook-form.com/api/useform/watch/
  // Callback version of watch.  It's your responsibility to unsubscribe when done.
  React.useEffect(() => {
    const subscription = watch((value, { name, type }) => console.log(value, name, type));
    return () => subscription.unsubscribe();
  }, [watch]);

File: src/modules/Builder/hooks/useActivitiesRedirection.ts [hunk 2]
```diff
@@ -12,15 +12,14 @@ export const useActivitiesRedirection = () => {
...
   const { activity } = useCurrentActivity();
   const isNewApplet = useCheckIfNewApplet();
-
   const { watch } = useFormContext();
 
-  const activities = watch('activities');
-  const shouldRedirect = (isNewApplet || activities?.length > 0) && activityId && !activity;
-
   useEffect(() => {
+    const activities = watch('activities');
```
Comment (alex-i-e) [inline src/modules/Builder/hooks/useActivitiesRedirection.ts:18]: please don't forget to unsubscribe then
https://www.react-hook-form.com/api/useform/watch/

**Review by alex-i-e — APPROVED** [review]

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**